### PR TITLE
Code refactoring & update extension compatible with Magento 2.4.3

### PR DESCRIPTION
**Pull request by andykim**
## Description

Code refactoring & update extension compatible with Magento 2.4.3

**Context:** 
- Many functions are duplicated for all payment methods and make the file is messy & not easy for maintenance. So that's why we need to make it shorter & reuse.
- Also need to write a unit test.
- The Abobe sent the warning email about delisting extension not support for magento 2.4.3 on the marketplace. So we need to upgrade the extension compatible with Magento 2.4.3.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Cosmetic change (text changing or color changing)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code (if necessary), particularly in hard-to-understand areas
- [ ] My changes generate no new vulnerabilities
- [ ] I have added tests that prove my fix is effective or that my feature works

## Screenshot

`put screenshot here if there are any ui changes`

### CODE REVIEW

File: Xendit/M2Invoice/etc/config.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
                 <xendit_url>https://tpi.xendit.co</xendit_url>
                 <ui_url>https://tpi-ui.xendit.co</ui_url>
                 <active>1</active>
-                <model>Xendit\M2Invoice\Model\Payment\Xendit</model>
+                <model>XenditFacade</model>
```
Comment (IreneGohtami) [inline Xendit/M2Invoice/etc/config.xml:11]: what's this?

File: Xendit/M2Invoice/Model/Payment/AbstractInvoice.php [hunk 2]
```diff
@@ -221,102 +156,24 @@ public function __construct(
...
-
-        try {
-            $response = $this->apiHelper->request($url, $method, null, null, null, $options);
-        } catch (\Exception $e) {
-            throw $e;
-        }
-
-        if (!isset($response['available_method'])) {
-            return [];
+        $amount = ceil($quote->getSubtotal() + $quote->getShippingAddress()->getShippingAmount());
```
Comment (IreneGohtami) [inline Xendit/M2Invoice/Model/Payment/AbstractInvoice.php:171]: @andykim we no longer need to round up the amount, invoice API will handle it. So here, we can remove the `ceil`.

File: Xendit/M2Invoice/etc/config.xml (hunk 1 above)
Comment (andykim) [reply Xendit/M2Invoice/etc/config.xml:11]: @IreneGohtami This  is the Payment Method Facade used when we create new payment method. I see it config on the di.xml file but did not use correct way in config.xml file. So I just put it on the correct way.

File: Xendit/M2Invoice/Model/Payment/AbstractInvoice.php (hunk 2 above)
Comment (andykim) [reply Xendit/M2Invoice/Model/Payment/AbstractInvoice.php:171]: @IreneGohtami ah. this one just used for checking the minimum & maximum amount, it stills not send to Xendit API on this step. But I will remove `ceil` on controller where we call to Xendit API to createInvoice.

Comment (andykim) [reply Xendit/M2Invoice/Model/Payment/AbstractInvoice.php:171]: @IreneGohtami I remove function `ceil` on order total before sending request to Invoice API on new commit. Please check it. Thanks

File: Xendit/M2Invoice/etc/config.xml (hunk 1 above)
Comment (andykim) [reply Xendit/M2Invoice/etc/config.xml]: @IreneGohtami I re-check and we don't need to use `XenditFacade` in this case. So I reverted it back `<model>Xendit\M2Invoice\Model\Payment\Xendit</model>`

**Review by IreneGohtami — APPROVED** [review]
LGTM

**Review by yanoandri — APPROVED** [review]

**Comment by rendegosling** [comment]
LGTM but here is my feedback:

1. PR is too big. Don't target to put all changes on 1 PR. smaller PRs are better and easier to review. Personally, I'll break this into (1) refactor, (2) upgrade version.
2. commit messages are good 👍 reveals intent.

Separate topic, can we add CI in Github Action to run the unit tests everytime we create a PR?

**Comment by andykim** [comment]
@rendegosling . Yaa. I will split it smaller PR on next time. 

About the CI in github action. I'm not sure we can because when run the unit tests it require some Magento config files. I will research about it and let you know

**Comment by rendegosling** [comment]
I think @IreneGohtami was able to use set up a different CI before for PHP. It would be good if we can use Buddy or GH action. I'll also check for other options

**Pull request merged**